src/components/DataDecorator/DataDecorator.js

```diff
@@ -59,7 +59,25 @@ class DataDecorator extends Component {
       scoreThresholds,
       scoreDescription,
       onContextMenu: propOnContextMenu,
+      midLineTruncation,
     } = this.props;
+    const truncationUtil = (inputText, maxLength, front, back) => {
```

Comment:
👏

Reply:
Would it make sense to have these changes in the `Decorator` also? I understand it's needed for the `Panel`, so perhaps there's a way to share the logic so there's parity between the `DataDecorator` and `Decorator`.

Reply:
I will let `Decorator` component handles the truncation work instead. And by doing so, both `DataDecorator` and `Decorator` can share the same API of truncation.